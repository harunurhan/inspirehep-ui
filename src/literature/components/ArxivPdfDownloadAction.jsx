import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ListItemAction from '../../common/components/ListItemAction';

class ArxivPdfDownloadAction extends Component {
  render() {
    const { arxivId } = this.props;
    const href = `//arxiv.org/pdf/${arxivId}`;
    return (
      <ListItemAction
        iconType="download"
        text="pdf"
        href={href}
        target="_blank"
      />
    );
  }
}

ArxivPdfDownloadAction.propTypes = {
  arxivId: PropTypes.string.isRequired,
};

export default ArxivPdfDownloadAction;
