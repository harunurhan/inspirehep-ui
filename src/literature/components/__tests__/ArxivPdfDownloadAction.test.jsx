import React from 'react';
import { shallow } from 'enzyme';

import ArxivPdfDownloadAction from '../ArxivPdfDownloadAction';

describe('ArxivPdfDownloadAction', () => {
  it('renders with arXiv id', () => {
    const wrapper = shallow((
      <ArxivPdfDownloadAction arxivId="123.12345" />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
