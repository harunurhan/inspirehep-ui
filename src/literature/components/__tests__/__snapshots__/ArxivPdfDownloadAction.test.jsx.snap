// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArxivPdfDownloadAction renders with arXiv id 1`] = `
<ListItemAction
  href="//arxiv.org/pdf/123.12345"
  iconType="download"
  target="_blank"
  text="pdf"
/>
`;
